### PR TITLE
Further fix: if the tunnel wasn't established, it's not in datagram_flows either

### PR DESCRIPTION
**Pull request by deweerdt**
This further fixes the case described in https://github.com/h2o/h2o/pull/2616

### CODE REVIEW

**Comment by kazuho** [comment]
Thank you for the PR. I believe we have to merge master as well?

**Comment by deweerdt** [comment]
master doesn't have bc46d90912902f3f7f8d1535d3bb9c1c4a58543f which contains `destroy_tunnel`

File: lib/http3/server.c [hunk 1]
```diff
@@ -396,7 +396,9 @@ static void destroy_tunnel(struct st_h2o_http3_server_stream_t *stream)
     struct st_h2o_http3_server_conn_t *conn = get_conn(stream);
     if (stream->tunnel->datagram_flow_id != UINT64_MAX) {
         khiter_t iter = kh_get(stream, conn->datagram_flows, stream->tunnel->datagram_flow_id);
-        assert(iter != kh_end(conn->datagram_flows));
+        /* the tunnel wasn't established yet */
+        if (iter == kh_end(conn->datagram_flows))
+            return;
         kh_del(stream, conn->datagram_flows, iter);
```
Comment (kazuho) [inline lib/http3/server.c:402]: I wonder if we should avoid early return here. The rationale is that the function is expected to dispose all the properties, where `datagram_flow_id` is just one among them. Therefore, I tend to believe that a statement (or a block) that disposes each element should not do an early return, and that applies to the last property as well for the sake of consistency.

**Review by kazuho — APPROVED** [review]
@deweerdt Sorry if I confused you, what I meant was that we'd have to merge master to kazuho/connect-udp to incorporate #2616. Am I correct? If I'm correct, please feel free to do the merge and push.

This PR by itself looks fine modulo an editorial suggestion below.

**Comment by deweerdt** [comment]
> @deweerdt Sorry if I confused you, what I meant was that we'd have to merge master to kazuho/connect-udp to incorporate #2616. Am I correct? If I'm correct, please feel free to do the merge and push.

Ah i see, will do. Thanks!

**Pull request merged**